### PR TITLE
Add an auto grading job/service for grading submissions.

### PR DESCRIPTION
**Pull request by lowjoel**
Depends on #568.

### CODE REVIEW

**Comment by lowjoel** [comment]
@allenwq should be fixed, thanks.

File: spec/services/course/assessment/answer/auto_grading_service_spec.rb [hunk 1]
```diff
@@ -3,23 +3,22 @@
...
         subject
 
-        expect(auto_grading).to be_graded
+        expect(auto_grading.answer).to be_graded
       end
     end
 
     describe '#grade' do
-      it 'sets the status to graded' do
+      it "sets the answer's status to graded" do
```
Comment (allenwq) [inline spec/services/course/assessment/answer/auto_grading_service_spec.rb]: We prefer `"sets the answer's status to graded"` or `'sets the answer\'s status to graded'` ? I remember once you mentioned to use second form.

**Comment by allenwq** [comment]
@lowjoel Coverage failed, I'll merge this after that.

**Comment by lowjoel** [comment]
I need to figure out a way to run the job such that that line is triggered. That line is only triggered when the job is run asynchronously, which is not the case in test.

File: spec/services/course/assessment/answer/auto_grading_service_spec.rb (hunk 1 above)
Comment (lowjoel) [inline spec/services/course/assessment/answer/auto_grading_service_spec.rb]: Yeah, hmm, I can't decide... we can normalise them all at once I guess. It's a simple grep.

Comment (allenwq) [inline spec/services/course/assessment/answer/auto_grading_service_spec.rb]: Haha, small issue, I'm fine with both

**Comment by lowjoel** [comment]
@allenwq got it!

**Comment by lowjoel** [comment]
Also, 377233239aa471546c24f091397c258d2003bffc adds support for a simple background thread job adapter, to simplify the development set up.

**Comment by allenwq** [comment]
@lowjoel Great, thanks !

**Pull request merged**